examples/type-level-boolean-programming

```diff
@@ -0,0 +1,38 @@
+/// A list that tracks whether its length is even or odd.
+enum EOList[a: Type, even: Bool] {
+    case EONil
+    case EOCons(a, EOList[a, not even])
+}
+
+/// Returns an empty and even list.
+def mkNilEO(): EOList[a, true] = EOList.EONil
+
+/// Attaches an element to the front of the list.
+def consEO(x: a, list: EOList[a, even]): EOList[a, not even] =
+    EOList.EOCons(x, list)
+
+/// Returns the head of the list.
+def headEO(list: EOList[a, even]): a = match list {
+    case EOList.EOCons(x, _) => x
+    case EOList.EONil        => bug!("Empty input not allowed")
+}
+
+/// Returns the tail of the list.
+def tailEO(list: EOList[a, even]): EOList[a, not even] = match list {
+    case EOList.EOCons(_, smallerList) => smallerList
+    case EOList.EONil                  => bug!("Empty input not allowed")
+}
+
+/// Applies a function `f` to each element of a list.
+def mapEO(f: a -> b, list: EOList[a, even]): EOList[b, even] = match list {
+    case EOList.EONil            => EOList.EONil
+    case EOList.EOCons(x, sList) => EOList.EOCons(f(x), mapEO(f, sList))
+}
+
+/// Prints both elements of a constructed list.
```

Comment:
Drop main

Reply:
I am willing to, but I just wanted to check - don't most of the examples have main functions? And in that case, should I preface each function name with _ to avoid unusage errors?